From a65243050c2c765b8c9edfdab9164c82c754c39b Mon Sep 17 00:00:00 2001
From: Alexander Bokovoy <abokovoy@redhat.com>
Date: Mon, 20 Jun 2016 21:41:35 +0300
Subject: [PATCH 2/4] Initialize ret before use

---
 src/back-sch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/back-sch.c b/src/back-sch.c
index ff8e8a2..bb2aa74 100644
--- a/src/back-sch.c
+++ b/src/back-sch.c
@@ -2376,7 +2376,7 @@ static int
 backend_extop_cb(Slapi_PBlock *pb)
 {
 	struct plugin_state *state;
-	int ret;
+	int ret = 0;
 	int i;
 	char *oid = NULL;
 	IFP fct = NULL;
-- 
2.7.4

