From 84c0767f794ca339275f9ec2d4936903d0ac3a42 Mon Sep 17 00:00:00 2001
From: Alexander Bokovoy <abokovoy@redhat.com>
Date: Mon, 20 Jun 2016 21:45:37 +0300
Subject: [PATCH 4/4] Declare int backend_init_extop for reuse in plug-sch.c

---
 src/back-sch.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/back-sch.h b/src/back-sch.h
index c15d1ed..1258ae0 100644
--- a/src/back-sch.h
+++ b/src/back-sch.h
@@ -164,6 +164,7 @@ bool_t backend_retrieve_from_nsswitch(struct backend_staged_search *staged,
 				      struct backend_search_cbdata *cbdata);
 
 int backend_sch_do_pam_auth(Slapi_PBlock *pb, const char *username);
+int backend_init_extop(Slapi_PBlock *pb, struct plugin_state *state);
 
 #ifdef USE_IPA_IDVIEWS
 void idview_get_overrides(struct backend_search_cbdata *cbdata);
-- 
2.7.4

